DataConnectors/SentinelOne/SentinelOneSentinelConnector/__init__.py

```diff
@@ -49,14 +57,43 @@ def generate_date(self):
             logging.info("There is no last time point, trying to get events for last day.")
             past_time = (current_time - datetime.timedelta(days=1)).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
         state.post(current_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"))
+        logging.info("Getting data from: " + past_time)
+        logging.info("To: " + current_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"))
         return (past_time, current_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"))
     
     def get_report(self, report_type_suffix, report_type_name, params = None):
         try:
             r = requests.get(self.domain + report_type_suffix, headers = self.header, params = params)
             if r.status_code == 200:
+                if("Threats" in report_type_name):
+                    self.get_threat_id(r.text)

```

Comment:
directly load the json from the request object rather than passing text
There is json method in request.

Reply:
I am using [this](https://stackoverflow.com/questions/16877422/whats-the-best-way-to-parse-a-json-response-from-the-requests-library) method. I need the JSON as dictionary later in the code to get the threat IDs.

Reply:
you are not using json method of request.
you could have passed the dictionary in the method rather than parsing it in the method and creating one.